gcp-lib.js

```diff
@@ -0,0 +1,60 @@
+/* eslint no-use-before-define: 0 */
+
+const _ = require("lodash");
+const GCCompute = require("@google-cloud/compute");
+const helpers = require("./helpers");
+
+const RESOURCE_OPERATIONS = {
+  create: _.partial(callMethod, "insert"),
+  get: async (client, request) => (await _.partial(callMethod, "get")(client, request))[0],
+  delete: _.partial(callMethod, "delete"),
+  listAsync: _.partial(callMethod, "listAsync"),
+  searchProjectsAsync: _.partial(callMethod, "searchProjectsAsync"),
+};
+
+async function callResourceOperation(
+  resourceOperation,
+  clientClass,
+  credentials,
+  project,
+  resource,
+  waitForOperation,
```

Comment:
I'm not a fan of this many parameters on the single function, but I don't have anything to offer instead, so it can stay until no other ideas appear.